newIDE/app/src/Course/LockedCourseChapterPreview.js

```diff
@@ -0,0 +1,218 @@
+import Window from '../Utils/Window';
+import PasswordPromptDialog from '../AssetStore/PasswordPromptDialog';
+
+const getYoutubeVideoIdFromUrl = (youtubeUrl: ?string): ?string => {
+  if (!youtubeUrl || !youtubeUrl.startsWith('https://youtu.be/')) return null;
+
+  try {
+    const url = new URL(youtubeUrl);
+
+    const lastPartOfUrl = url.pathname.split('/').pop();
+    if (!lastPartOfUrl || !lastPartOfUrl.length) {
+      console.error(`The video URL is badly formatted ${youtubeUrl}`);
+      return null;
+    }
+    return lastPartOfUrl;
+  } catch (error) {
+    console.error(`Could not parse youtube url ${youtubeUrl}:`, error);
+    return null;
+  }
+};
+
+const styles = {
+  videoAndMaterialsContainer: {
+    display: 'flex',
+    marginTop: 8,
+    gap: 8,
+    alignItems: 'stretch',
+    flexWrap: 'wrap',
+    marginBottom: 8,
+    flex: 1,
+    minWidth: 0,
+  },
+  lockedChapterTextContainer: {
+    // Similar to Line component but without the minWidth:0 that somehow
+    // prevents container to wrap when overflowing.
+    display: 'flex',
+    flex: 1,
+    minHeight: 0,
+  },
+  videoContainer: {
+    flex: 2,
+    minWidth: 300,
+    display: 'flex',
+    position: 'relative',
+  },
+  videoThumbnail: {
+    flex: 1,
+    aspectRatio: '16 / 9',
+    objectFit: 'cover',
+    width: '100%',
+    borderRadius: 4,
+  },
+  lockerImage: { height: 60, width: 60 },
+  lockedOverlay: {
+    position: 'absolute',
+    background: 'rgba(0, 0, 0, 0.6)',
+    borderRadius: 4,
+    inset: 0,
+    height: `100%`,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    color: 'white', // Force text/icon color since it's on a dark overlay.
+  },
+  sideBar: { padding: 16, display: 'flex' },
+};
+
+const LockedOverlay = () => (
+  <div style={styles.lockedOverlay}>
+    <Lock style={styles.lockerImage} />
+  </div>
+);
+
+type Props = {|
+  courseChapter: LockedVideoBasedCourseChapter | LockedTextBasedCourseChapter,
+  onBuyWithCredits: (CourseChapter, string) => Promise<void>,
+|};
+
+const LockedCourseChapterPreview = React.forwardRef<Props, HTMLDivElement>(
+  ({ courseChapter, onBuyWithCredits }, ref) => {
+    const { openSubscriptionDialog } = React.useContext(
+      SubscriptionSuggestionContext
+    );
+    const [
+      displayPasswordPrompt,
+      setDisplayPasswordPrompt,
+    ] = React.useState<boolean>(false);
+    const { windowSize } = useResponsiveWindowSize();
+    const [password, setPassword] = React.useState<string>('');
+    const youtubeVideoId = courseChapter.videoUrl
+      ? getYoutubeVideoIdFromUrl(courseChapter.videoUrl)
+      : null;
+    const [isPurchasing, setIsPurchasing] = React.useState<boolean>(false);
+
+    const onClickBuyWithCredits = React.useCallback(
+      async () => {
+        if (!courseChapter.isLocked) return;
+
+        setDisplayPasswordPrompt(false);
+        setIsPurchasing(true);
+        try {
+          await onBuyWithCredits(courseChapter, password);
```

Comment:
do we need to catch and display an error if it fails?

Reply:
Good idea, on it

Reply:
<img width="406" alt="image" src="https://github.com/user-attachments/assets/762e52e7-0087-44b3-8a6e-5777c2edd771" />

Reply:
Nice!